dbmongo/js/reduce.algo2/cibleApprentissage.ts

```diff
@@ -15,7 +28,7 @@ function cibleApprentissage(output_indexed, n_months) {
       ),
     }
     return m
-  }, {})
+  }, {} as Record<string, Record<string, unknown>>)
```

Comment:
Je dirais plutôt:
```suggestion
  }, {} as Record<Periode, Record<string, boolean>>)
```
voire `Record<Periode, Record<"outcome", boolean>>` mais je ne sais pas si cette notation est autorisée...

Reply:
Je ne peux pas mettre `Periode` ici en l'état, car le type `Periode` défini en global est un alias du type `Date`, or le type des clés d'un `Record` doit dériver de `string | number | symbol` (ce qui n'est pas le cas de `Date`).

Veux tu que je définisse un type `Periode` local qui serait un alias de `string` ?

Reply:
Dans `globals.ts` on a:
`type Periode = string // Date`
Je loupe qqchose ?

Reply:
effectivement, j'ai du faire une mauvaise manip quand j'ai testé => j'applique le code que j'avais suggéré dans mon commentaire précédent, sans la (re)définition du type `Periode`.